query-algebrizer/src/errors.rs

```diff
@@ -40,6 +40,12 @@ error_chain! {
             description("invalid argument")
             display("invalid argument to {}: expected numeric in position {}.", function, position)
         }
+
+        NonMatchingVariablesInOrClause {
+            // TODO: flesh out.
+            description("non-matching variables in 'or' clause")
```

Comment:
I think you're asking about wording.

IMO there's not a lot of clarity in the distinction, if any, but:

- "Unmatched variables" kinda means we were looking for one or two and didn't get them.
- "Mismatched variables" means we were combining two things and they didn't match. A mismatch is bidirectional.
- "Non-matching variables" implies quite weakly that some expected property didn't hold.

That's the weak reasoning behind this wording: this error means that the expected property — that every arm in the `or` uses all of the variables in the rule vars, which might themselves be derived from the arms — doesn't hold. There can be an arbitrary number of arms.